### PR TITLE
Manage Roles with SCIM 2.0 and SOAP services Product Scenario TestCases

### PR DESCRIPTION
**Pull request by sarubi**
Covered scenarios:
1. Update a provisioned user and assign to a new role and assert if the role is assigned to the user
2. [N] Rename an existing role to a name of an existing role
3. Provision a user with admin rights
4. Add a primary role without adding any permissions
5. Add a primary role with few permissions and assign to an existing user
6. Add a primary role with few permissions by not assign it to a user
7. Add an internal role without adding any permissions
8. Add an internal role with few permissions and assign to an existing user
9. Add an internal role with few permissions without assign it to a user

### CODE REVIEW

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/java/org/wso2/identity/scenarios/test/services/ManageInternalRolesWithPermissionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        this.internalRoleName = internalRoleName;
+        this.userList = userList;
+    }
+
+    @DataProvider(name = "manageRolesConfigProvider")
+    private static Object[][] manageRolesConfigProvider() {
+
+        return new Object[][] {
+                {
+                        "engineer", new String[] { "user3", "user4" }
```
Comment (Kavindu-Dodan) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/java/org/wso2/identity/scenarios/test/services/ManageInternalRolesWithPermissionTestCase.java]: Can we check the formatting here. Seems indentations are not correct

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/java/org/wso2/identity/scenarios/test/services/ManagePrimaryRolesWithPermissionTestCase.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        this.roleName = roleName;
+        this.userList = userList;
+    }
+
+    @DataProvider(name = "manageRolesConfigProvider")
+    private static Object[][] manageRolesConfigProvider() {
+
+        return new Object[][] {
+                {
+                        "Manager", new String[] { "user1", "user2" }
```
Comment (Kavindu-Dodan) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/java/org/wso2/identity/scenarios/test/services/ManagePrimaryRolesWithPermissionTestCase.java]: Same as previous, check indentation

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/resources/testng.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    </listeners>
+
+    <test name="is-groups-mgt-soap-services" preserve-order="true" parallel="false">
+        <classes>
+            <class name="org.wso2.identity.scenarios.test.services.ManagePrimaryRolesWithPermissionTestCase"/>
+            <class name="org.wso2.identity.scenarios.test.services.ManageInternalRolesWithPermissionTestCase"/>
+            <class name="org.wso2.identity.scenarios.test.services.ProvisionUserWithAdminRights"/>
+        </classes>
+    </test>
+</suite>
```
Comment (GDRDABARERA) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/resources/testng.xml]: any reason to add the blow empty file?

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        groupId = returnedGroupJSON.get(ID_ATTRIBUTE).toString();
+        assertNotNull(groupId, "SCIM2 group id not available in the response.");
+        EntityUtils.consume(response.getEntity());
+    }
+
+    @Test(dependsOnMethods = "testCreateGroupWithMembers")
+    private void testDeleteUser() throws Exception {
+
+        for (Map.Entry<String, String> user : users.entrySet()) {
+            HttpResponse response = scim2Client.deleteUser(client, user.getValue(), ADMIN_USERNAME, ADMIN_PASSWORD);
```
Comment (GDRDABARERA) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java]: Isn't this line exceeded 120 characters?

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                    "Failed to delete the user");
+            EntityUtils.consume(response.getEntity());
+        }
+    }
+
+    @Test(dependsOnMethods = "testDeleteUser")
+    public void testDeleteCreatedGroup() throws Exception {
+
+        HttpResponse response = scim2Client.deleteGroup(client, groupId, ADMIN_USERNAME, ADMIN_PASSWORD);
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_NO_CONTENT, "Failed to delete the group");
```
Comment (GDRDABARERA) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java]: format the line to 120

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RenameExistingRoleToNameOfAnExistingRoleSCIM2TestCase.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            EntityUtils.consume(response.getEntity());
+        }
+    }
+
+    @Test(dependsOnMethods = "testCreateGroups")
+    public void testGetGroups() throws Exception {
+
+        for (int i = 0; i < inputFileName.length; i++) {
+            HttpResponse response = scim2Client.getGroup(client, groupId[i], username, password);
+            assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK, "Unable to retrieve the group");
```
Comment (GDRDABARERA) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RenameExistingRoleToNameOfAnExistingRoleSCIM2TestCase.java]: Format lines

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/UpdateRoleSCIM2TestCase.java [hunk 7]
```diff
@@ -89,8 +90,7 @@ public void testUpdatePrimaryGroup() throws Exception {
 
         JSONObject groupJSON = scim2Client.getRoleJSON("scim2UpdateGroup.json");
         HttpResponse response = scim2Client.updateGroup(client, groupJSON, groupId, username, password);
-        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK,
-                "Group has been updated successfully");
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK, "Group has been updated successfully");
```
Comment (GDRDABARERA) [inline product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/UpdateRoleSCIM2TestCase.java]: format lines

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java (hunk 4 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java]: No..

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java (hunk 5 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/ManageRolesWithUsersSCIM2TestCase.java]: exceeded to 120

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RenameExistingRoleToNameOfAnExistingRoleSCIM2TestCase.java (hunk 6 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/RenameExistingRoleToNameOfAnExistingRoleSCIM2TestCase.java]: Not exceeded to 120

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/UpdateRoleSCIM2TestCase.java (hunk 7 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.1-manage-roles-with-scim2/src/test/java/org.wso2.identity.scenarios.test.scim2/UpdateRoleSCIM2TestCase.java]: Screenshot from 2019-09-25 19-11-12

File: product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/resources/testng.xml (hunk 3 above)
Comment (sarubi) [reply product-scenarios/1-manage-users-and-roles/1.1-admin-user-manage-roles/1.1.3-manage-roles-with-soap-services/src/test/resources/testng.xml]: test.sh already existing one, no changes done...

**Pull request merged**